### PR TITLE
feat(Stories): Add guidance for component and token usage

### LINKED ISSUE #3750: Add design tokens to Storybook
### Description

Design tokens for color, spacing and typography should be included in the storybook and built from the actual Design system.

### PR DESCRIPTION
**Pull request by markhigham**
## Related issue

Closes #3750

## Overview

Add guidance for component and token usage to the storybook. Token examples are based on the actual design tokens.

## Reason
We wanted to reduce the cost / overhead of maintaining a separate DS website _and_ the storybook.
It was important to generate the tokens docs from the _actual code_ 

## Work carried out

- [x] Enable mdx for storybook
- [x] Reorganise existing stories into `Components` folder
- [x] Create new tokens stories
- [x] Reorganise forms stories and import guidance from docs site 
- [x] Create overall guidance from existing site content.   
- [x] Fix broken e2e tests which were dependent on storybook paths

### CODE REVIEW

**Comment by m7kvqbe1** [comment]
Can we change these commit message types from `feat` and `fix` to `docs` where they are Storybook related.

File: packages/react-component-library/.storybook/RoyalNavyTheme.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import { create } from '@storybook/theming/create'
+
+export default create({
+  base: 'light',
```
Comment (m7kvqbe1) [inline packages/react-component-library/.storybook/RoyalNavyTheme.js]: Can we remove this please - my system setting is dark and this makes it default to light instead on load.

File: packages/react-component-library/.storybook/preview.js [hunk 2]
```diff
@@ -31,6 +31,23 @@ export const parameters = {
...
+        'Designers',
+        'Tokens',
+        'Forms',
+        [
+          'Introduction',
+          'Anatomy of a form',
+          'General form standards',
+          'Form layout',
+          'Form alignment and spacing',
+          'Submit, validate and show form errors',
```
Comment (m7kvqbe1) [inline packages/react-component-library/.storybook/preview.js]: I wonder if we can come up with a more succinct title for this - it takes up a lot of space in the sidebar.

File: packages/react-component-library/src/forms/alignment.mdx [hunk 3]
```diff
@@ -0,0 +1,52 @@
+import { Meta, Source } from '@storybook/blocks'
+
+<Meta title="Forms/Form alignment and spacing" />
+
+# Form alignment and spacing
```
Comment (m7kvqbe1) [inline packages/react-component-library/src/forms/alignment.mdx:5]: Change to: Alignment and spacing

File: packages/react-component-library/src/forms/alignment.mdx [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+###### 7.2 You must not use more than four different widths for form components.
+> While input field widths must reflect the expected input length, too many distinct widths result in visual ‘noise’ and have the opposite effect than intended. To maintain visual balance use components that correspond to 100%, 75%, 50% or 25% of your form width, with the necessary spacing between them as described in the next section.
+
+<img src="/alignment_do.svg" alt="form alignment" />
+
+<img src="/alignment_dont.svg" alt="form alignment what not to do" />
+
+## 8. Spacing
+###### 8.0 You *must* ensure all form components are spaced using the margins specified.
+###### 8.1 The margin between your form and its container or nearest neighbour *must not* be smaller than 20 pixels. Where possible, you should aim for a margin of at least 32 pixels. Different contexts may require different margins. In exceptional cases (eg. a small number of fields within a complex interface), you may use a small margin, but no smaller than 20 pixels. Forms may appear as dedicated pages or within a Dialog, a Drawer or a Modal.
```
Comment (m7kvqbe1) [inline packages/react-component-library/src/forms/alignment.mdx]: Missing a line break and > here

```suggestion
###### 8.1 The margin between your form and its container or nearest neighbour *must not* be smaller than 20 pixels. 
>Where possible, you should aim for a margin of at least 32 pixels. Different contexts may require different margins. In exceptional cases (eg. a small number of fields within a complex interface), you may use a small margin, but no smaller than 20 pixels. Forms may appear as dedicated pages or within a Dialog, a Drawer or a Modal.
```

**Comment by m7kvqbe1** [comment]
The font used in the SVG graphics appears to be incorrect - it should be using the one from the fonts package.

File: packages/react-component-library/src/forms/submit.mdx [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+###### 11.1 You should prevent user errors by offering suggestions, utilising constraints, and being appropriately flexible.
+Both expert users familiar with the process at hand and novice users still learning are prone to making mistakes,
+especially under pressure. When users move between systems - and especially systems that use different
+conventions - the chances of making a mistake increase.
+For example, entering a date in a `mm/dd/yyyy` format when the system expects a `dd/mm/yyyy` format
+or typing a time using a 12-hour clock when the system uses a 24-hour format are both difficult
+to validate and could have severe consequences.
+
+We can reduce such errors using good design. For example, when asking the user to enter a date,
+show a “(dd/mm/yyyy)” note. When choosing a time, you can immediately display the entered time into 24-hour
```
Comment (m7kvqbe1) [inline packages/react-component-library/src/forms/submit.mdx]: Can we find and replace these dodgy double quotes chracters?

```suggestion
show a "(dd/mm/yyyy)" note. When choosing a time, you can immediately display the entered time into 24-hour
```

File: packages/react-component-library/src/intro/developer-guide.mdx [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+  return (
+    <GlobalStyleProvider>
+      <Button variant="primary">
+        Hello, World!
+      </Button>
+    </GlobalStyleProvider>
+  )
+}
+
+ReactDOM.render(<App />, document.querySelector('#app'))`} />
```
Comment (m7kvqbe1) [inline packages/react-component-library/src/intro/developer-guide.mdx]: This needs updaintg to use the new react-dom API:

```typescript
import { createRoot } from "react-dom/client";

const root = createRoot(rootElement);

root.render(
  // ...
)

```

File: packages/react-component-library/src/intro/developer-guide.mdx [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+
+---
+
+## More examples
+
+<iframe src="https://codesandbox.io/embed/kpx2z9?view=Preview&module=%2Fsrc%2Findex.js"
```
Comment (m7kvqbe1) [inline packages/react-component-library/src/intro/developer-guide.mdx:47]: Example is throwing a console error:

```
Warning: React instrumentation encountered an error: TypeError: ie.patchConsoleForStrictMode is not a function
SampleApp@https://kpx2z9.csb.app/src/sample.jsx:15:59
div
O@https://kpx2z9.csb.app/node_modules/styled-components/dist/styled-components.browser.esm.js:682:7
```

File: packages/react-component-library/src/tokens/SpacingScaleTable.tsx [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+
+export const SpacingScaleTable = () => {
+  return (
+    <StyledTable>
+      {keys.map((key) => (
+        <StyledRow key={key}>
+          <div
+            style={{
+              backgroundColor: '#FFF',
+              border: `dotted 1px ${color('neutral', '300')}`,
```
Comment (m7kvqbe1) [inline packages/react-component-library/src/tokens/SpacingScaleTable.tsx]:
```suggestion
              border: `1px solid ${color('neutral', '300')}`,
              text-indent: ${spacing('4')};
```

File: packages/react-component-library/src/tokens/SpacingScaleTable.tsx [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+            &nbsp;{key}:&nbsp;{spacing(key)}
+          </div>
+        </StyledRow>
+      ))}
+    </StyledTable>
+  )
+}
+
+export const ExampleComponent = () => {
+  /*  This is a contrived example to show individual usage
```
Comment (m7kvqbe1) [inline packages/react-component-library/src/tokens/SpacingScaleTable.tsx]:
```suggestion
/**
 * This is a contrived example to show individual usage.
 */
```

File: packages/react-component-library/src/tokens/TypographyTable.tsx [hunk 10]
```diff
@@ -0,0 +1,48 @@
+import React from 'react'
+import { selectors, TypographySize } from '@royalnavy/design-tokens'
+import styled from 'styled-components'
+
+const { fontSize, color, spacing } = selectors
+
+const tokens = [
```
Comment (m7kvqbe1) [inline packages/react-component-library/src/tokens/TypographyTable.tsx:7]: Not for now but I will have a look to see how we can avoid this being duplicated and hard-coded.

We should be able to change the tokens in a single place and have Storybook examples reflect that.

File: packages/react-component-library/src/tokens/colors.mdx [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+</ColorPalette>
+
+---
+
+## Usage
+<Source dark code={`import { selectors } from '@royalnavy/design-tokens'
+const { color } = selectors
+
+<div style={{ backgroundColor: color('action', '400') }}>
+  Hello colo(u)rful world
```
Comment (m7kvqbe1) [inline packages/react-component-library/src/tokens/colors.mdx]:
```suggestion
  Hello, colo(u)rful World!
```

File: packages/react-component-library/src/tokens/spacing.mdx [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+
+## Usage
+
+<Source dark code={`# Typescript usage of spacing
+import { selectors } from '@royalnavy/design-tokens'
+
+const { spacing } = selectors
+
+export const ExampleComponent = () => {
+  /*  This is a contrived example to show individual usage
```
Comment (m7kvqbe1) [inline packages/react-component-library/src/tokens/spacing.mdx:26]:
```suggestion
  /**
   * This is a contrived example to show individual usage.
   */
```

File: packages/react-component-library/src/tokens/typography.mdx [hunk 13]
```diff
@@ -0,0 +1,40 @@
+import { Meta, Source } from '@storybook/blocks'
+import { TypographyTable } from './TypographyTable'
```
Comment (m7kvqbe1) [inline packages/react-component-library/src/tokens/typography.mdx]: Missing line break.

File: packages/react-component-library/src/tokens/typography.mdx [hunk 14]
```diff
@@ -0,0 +1,40 @@
...
+
+
```
Comment (m7kvqbe1) [inline packages/react-component-library/src/tokens/typography.mdx:4]: Extra redudant line breaks.

File: packages/react-component-library/src/tokens/typography.mdx [hunk 15]
```diff
@@ -0,0 +1,40 @@
...
+
+<Meta title="Tokens/Typography" />
+
+# Typography
+
+All Royal Navy Design System applications are set in [Lato](https://www.latofonts.com/).
```
Comment (m7kvqbe1) [inline packages/react-component-library/src/tokens/typography.mdx:10]: This is a dead link to a broken website - can we remove please.

**Comment by markhigham** [comment]
> The font used in the SVG graphics appears to be incorrect - it should be using the one from the fonts package.

The Lato font stack isn't readily available in the mdx files. I used the system stack as an interim measure. We can take a look at how to make this consistent in a subsequent release. 

It's better than the current site which is times new roman

File: packages/react-component-library/.storybook/RoyalNavyTheme.js (hunk 1 above)
Comment (markhigham) [reply packages/react-component-library/.storybook/RoyalNavyTheme.js]: Done

File: packages/react-component-library/src/tokens/TypographyTable.tsx (hunk 10 above)
Comment (markhigham) [reply packages/react-component-library/src/tokens/TypographyTable.tsx:7]: #3758

File: packages/react-component-library/src/tokens/typography.mdx (hunk 15 above)
Comment (markhigham) [reply packages/react-component-library/src/tokens/typography.mdx:10]: The link works for me?

Comment (m7kvqbe1) [reply packages/react-component-library/src/tokens/typography.mdx:10]: Must have just been down when I hit it.

**Review by m7kvqbe1 — APPROVED** [review]

**Pull request merged**